src/plugins.c

```diff
@@ -830,6 +1000,52 @@ static gboolean check_plugin_path(const gchar *fname)
 }
 
 
+/* Retuns NULL if this ain't a plugin,
+ * otherwise it returns the appropriate PluginProxy instance to load it */
+static PluginProxy* is_plugin(const gchar *file)
+{
+	PluginProxy *proxy;
+	const gchar *tmp;
+	gint i;
+
+	/* extract file extension to avoid g_str_has_suffix() in the loop */
+	tmp = (const gchar *)g_utf8_strrchr(file, -1, '.');
+	if (tmp == NULL)
+		return FALSE;
+	/* ensure the dot is really part of the filename */
+	else if (strchr(tmp, G_DIR_SEPARATOR) != NULL)
```

Comment:
Oh look. I used strchr(). I'm inconsistent with myself :/

Regarding the /, do we have to do this everywhere? I didn't notice an #ifdef hell about that yet